app/drivers/gpio/max7318/Kconfig

```diff
@@ -0,0 +1,21 @@
+# MAX7318 GPIO configuration options
+
+# Copyright (c) 2022 The ZMK Contributors
+# SPDX-License-Identifier: MIT
+
+menuconfig GPIO_MAX7318
+	bool "MAX7318 I2C-based GPIO chip"
+	depends on I2C
+	select HAS_DTS_GPIO
```

Comment:
See the 595 gpio driver I just merged for an example of how to auto-enable this if found in the DT.

Reply:
yep, copied it over.